subprojects/kotlin-dsl-plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/precompiled/PrecompiledScriptPlugins.kt

```diff
@@ -78,4 +78,5 @@ val Project.sourceSets
 
 private
 val SourceSet.kotlin: SourceDirectorySet
-    get() = withConvention(KotlinSourceSet::class) { kotlin }
+    // TODO coordinate with the kotlin-gradle-plugin's authors to migrate from conventions to extensions
```

Comment:
💄 I would remove that comment and track this elsewhere

Reply:
Ok, I'll remove the comment and follow-up elsewhere.